src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -518,6 +574,7 @@ private static void InsertAndFreePrologue(ArrayBuilder<BoundStatement> result, A
         /// <param name="F">A function that computes the translation of the node.  It receives lists of added statements and added symbols</param>
         /// <returns>The translated statement, as returned from F</returns>
         private T IntroduceFrame<T>(BoundNode node, LambdaFrame frame, Func<ArrayBuilder<BoundExpression>, ArrayBuilder<LocalSymbol>, T> F)
+            where T : BoundNode
```

Comment:
Can we replace `T` with `BoundNode`, as in the VB implementation? It looks like all uses of `IntroduceFrame<T>` are in expressions that return a `BoundNode` rather than a derived type.